### ISSUE TITLE
Long paragraphs lead to sluggish editing

### ISSUE THREAD

**Issue by sitase** [glushchenko/fsnotes#1086]
When you type long paragraphs fsnotes reacts slower and slower to keypresses.

To reproduce, open a note and type a long paragraph (about 2300 characters makes this clear). The characters will appear noticeably sluggish.

If you start a new paragraph, speed is back to normal.

Expected: There  should not be any noticeable differences in processing key presses whether the paragraph is long or short.

MacOS Big Sur 11.2.1
FSNotes 4.2.2 (378)

**Comment by glushchenko** [comment]
This is should be slower and slower by design. More chars in paragraph -> more textprocessor loading (every paragraph rescans on char typing).

**Comment by pdanford** [comment]
Out of curiosity, does unchecking "Code block live highlighting (app restart required)" have any effect on the sluggishness?

**Comment by phocean** [comment]
> Out of curiosity, does unchecking "Code block live highlighting (app restart required)" have any effect on the sluggishness?

Yes, immediate impact. Sluggishness has been caused by that feature in my case.

**Comment by glushchenko** [comment]
https://github.com/glushchenko/fsnotes/releases/tag/7.0.0b1
